src/script/standard.cpp

```diff
@@ -44,33 +44,177 @@ const char *GetTxnOutputType(txnouttype t)
+static bool MatchLabelPublic(const CScript &script, std::vector<valtype> &dataCarriage)
 {
-    // Templates
-    static multimap<txnouttype, CScript> mTemplates = {
+    // LabelPublc OP_RETURN data size format small
+    // Template: "CScript() << OP_RETURN << OP_BIGINTEGER << OP_DATA"
+    // data deve contenere il OP_DATA
+    if (script.size() < 1 || script[0] != OP_RETURN || !script.IsPushOnly(script.begin() + 1))
+    {
+        return false;
+    }
 
-        // Standard tx, sender provides pubkey, receiver adds signature
-        {TX_PUBKEY, CScript() << OP_PUBKEY << OP_CHECKSIG},
+    valtype data;
+    opcodetype opcode;
+    CScript::const_iterator s = script.begin() + 1;
+    script.GetOp(s, opcode, data);
 
-        // Bitcoin address tx, sender provides hash of pubkey, receiver provides signature and pubkey
-        {TX_PUBKEYHASH, CScript() << OP_DUP << OP_HASH160 << OP_PUBKEYHASH << OP_EQUALVERIFY << OP_CHECKSIG},
+    try
+    {
+        CScriptNum dataId(data, true, 5);
+        dataCarriage.emplace_back(data);
+    }
+    catch (scriptnum_error)
+    {
+        return false;
+    }
+
+    if (script.GetOp(s, opcode, data))
+    {
+        dataCarriage.emplace_back(std::move(data));
+        return true;
+    }
 
-        // Sender provides N pubkeys, receivers provides M signatures
-        {TX_MULTISIG, CScript() << OP_SMALLINTEGER << OP_PUBKEYS << OP_SMALLINTEGER << OP_CHECKMULTISIG},
+    return false;
+}
+
+static bool MatchFreezeCLTV(const CScript &script, std::vector<valtype> &pubkeys)
+{
+    // Freeze tx using CLTV ; nFreezeLockTime CLTV DROP (0x21 pubkeys) checksig
+    // {TX_CLTV, CScript() << OP_BIGINTEGER << OP_CHECKLOCKTIMEVERIFY << OP_DROP << OP_PUBKEYS << OP_CHECKSIG},
+
+    if ((script.size() < 1) || script.back() != OP_CHECKSIG)
+    {
+        return false;
+    }
 
-        // Freeze tx using CLTV ; nFreezeLockTime CLTV DROP (0x21 pubkeys) checksig
-        {TX_CLTV, CScript() << OP_BIGINTEGER << OP_CHECKLOCKTIMEVERIFY << OP_DROP << OP_PUBKEYS << OP_CHECKSIG},
+    valtype data;
+    opcodetype opcode;
+    CScript::const_iterator s = script.begin();
+    script.GetOp(s, opcode, data);
 
-        // LabelPublc OP_RETURN data size format small
-        {TX_LABELPUBLIC, CScript() << OP_RETURN << OP_BIGINTEGER << OP_DATA}
+    try
+    {
+        CScriptNum nLockFreezeTime(data, true, 5);
+        pubkeys.emplace_back(data);
+    }
+    catch (scriptnum_error)
+    {
+        return false;
+    }
 
-    };
+    CScriptNum nLockFreezeTime(data, true, 5);
+    uint8_t pos = 5;
+    if (nLockFreezeTime < LOCKTIME_THRESHOLD)
+    {
+        pos = 4;
+    }
+    if ((script[pos] != OP_CHECKLOCKTIMEVERIFY) || (script[pos + 1] != OP_DROP))
+    {
+        return false;
+    }
 
+    // starting from pubkeys (4/5 byte nlock time + 1 OP_CLTV + 1 OP_DROP)
+    CScript::const_iterator it = script.begin() + pos + 2;
```

Comment:
here too, just use s + 2, or actually get rid of it and use s = s + 2

Reply:
done